File: Harish Kumar/Array and Strings/Minimum Swaps to Make Strings Equal/Minimum Swaps to Make Strings Equal.java
class Solution {
    public int minimumSwap(String s1, String s2) {
        int x1 = 0, y1 = 0, x2 = 0, y2 = 0;
        
        for (int i = 0; i < s1.length (); i++) {
            char c1 = s1.charAt (i);
            char c2 = s2.charAt (i);
            
            if (c1 != c2) {
                if (c1 == 'x') {
                    ++x1;
                }
                else {
                    ++y1;
                }
                
                if (c2 == 'x') {
                    ++x2;
                }
                else {
                    ++y2;
                }
            }
        }
        
        if ((x1 + x2) % 2 != 0 || (y1 + y2) % 2 != 0) {
            return -1;
        }
        
        return x1 % 2 == 0 ? (x1 + y1) / 2 : (x1 + y1) / 2 + 1;
    }
}

File: Ankit Arora/Arrays and Strings/MinimumSwapstomakestringsequal.cpp
class Solution {
public:
    int minimumSwap(string s1, string s2) {
        int c1=0 , c2=0 ;
        for(int i=0  ; i<s1.length() ; i++)
        {
            if(s1[i] != s2[i])
            {
                if(s1[i]=='x') c1++;
                else c2++;
            }
        }
        if(c1%2==0 and c2%2==0) return (c1+c2)/2;
        else if (c1%2==1 and c2%2==1) return (c1+c2)/2 + 1;
        return -1;

    }
};
